libnmstate/ifaces/ifaces.py

```diff
@@ -165,6 +166,35 @@ def _validate_ovs_patch_peers(self):
                             " patch port"
                         )
 
+    def _validate_vlan_mtu_lt_base_mtu(self):
+        for iface in self._ifaces.values():
+            if iface.iface_type == InterfaceType.VLAN:
+                if iface.state == InterfaceState.UP:
+                    base_iface_name = iface._vlan_config.get(VLAN.BASE_IFACE)
+                    # check if base_iface is in desired state.
+                    # if not then get from current state
+                    base_iface = self._ifaces.get(base_iface_name)
+                    if not base_iface:
+                        base_iface = self.cur_ifaces.get(base_iface_name)
+                        if not base_iface:
+                            raise ValueError(
+                                f"base interface {base_iface_name} for vlan \
+                                                {iface.name} does not exist"
+                            )
+
+                    base_iface_mtu = base_iface.mtu
+                    vlan_mtu = iface.mtu
+                    if (
+                        vlan_mtu
+                        and base_iface_mtu
```

Comment:
When base_iface/parent interface does not have MTU defined while VLAN/VxLAN does, we should set parent interface MTU as vlan's.
But it could be done in another PR if you don't want to do it.

Reply:
I have added this in the validator itself. 
is that fine or should it be added somewhere else?